src/params.js

```diff
@@ -67,3 +67,52 @@ export function useParams() {
     const params = useParamsStore().state
     return { params }
 }
+
+// For Pitcher Connect initialization
+window.setMainNav = function(lastViewedCategory) {
+    window.lastViewedCategory = lastViewedCategory
+}
+
+window.gotJSON = function(serverJSONV, documentPathV) {
+    try {
+        window.documentPath = documentPathV
+        window.serverJSON = JSON.parse(serverJSONV)
+        window.serverJSON.files.reverse()
+        window.appID = window.serverJSON.appID
+    } catch (e) {
+        window.Ti.App.fireEvent('Error', e)
+    }
+}
+
+window.loadPresentations = function() {}
+window.showUI = function() {}
+
+export function loadServerJSON() {
+    return new Promise(resolve => {
+        if (process.env.VUE_APP_PARAMS) {
+            // for testing
+            // let preParams = JSON.parse(process.env.VUE_APP_PARAMS)
+            // store.patch(preParams)
+            // resolve(store.state)
+            /*
+              TO DO:
+              Remove later
+            */
+            resolve(undefined)
+        } else if (typeof window.serverJSON === 'undefined') {
+            let count = 0
+            let interval = setInterval(() => {
+                count++
+                if (typeof window.serverJSON !== 'undefined' || count === 10) {
+                    clearInterval(interval)
+                    // store.patch(window.params)
```

Comment:
instead of returning only i would store it.

Reply:
probably in the same params Store...

Reply:
this way we can use the same data there.... if you are logged in for example and have it always at the same location?

Reply:
Then i would probably make one init() function that looks if getParams or serverJSON is called... if one of them is called... store it in params store and return